rclcpp/include/rclcpp/parameter.hpp

```diff
@@ -56,31 +57,43 @@ class ParameterVariant
   {
     value_.parameter_type = rcl_interfaces::ParameterType::PARAMETER_NOT_SET;
   }
-  ParameterVariant(const std::string & name, const bool bool_value)
+  explicit ParameterVariant(const std::string & name, const bool bool_value)
   : name_(name)
   {
     value_.bool_value = bool_value;
     value_.parameter_type = rcl_interfaces::ParameterType::PARAMETER_BOOL;
   }
-  ParameterVariant(const std::string & name, const int64_t int_value)
+  explicit ParameterVariant(const std::string & name, const int int_value)
   : name_(name)
   {
     value_.integer_value = int_value;
     value_.parameter_type = rcl_interfaces::ParameterType::PARAMETER_INTEGER;
   }
-  ParameterVariant(const std::string & name, const double double_value)
+  explicit ParameterVariant(const std::string & name, const int64_t int_value)
+  : name_(name)
+  {
+    value_.integer_value = int_value;
+    value_.parameter_type = rcl_interfaces::ParameterType::PARAMETER_INTEGER;
+  }
+  explicit ParameterVariant(const std::string & name, const float double_value)
   : name_(name)
   {
     value_.double_value = double_value;
     value_.parameter_type = rcl_interfaces::ParameterType::PARAMETER_DOUBLE;
   }
-  ParameterVariant(const std::string & name, const std::string & string_value)
+  explicit ParameterVariant(const std::string & name, const double double_value)
+  : name_(name)
+  {
+    value_.double_value = double_value;
+    value_.parameter_type = rcl_interfaces::ParameterType::PARAMETER_DOUBLE;
+  }
+  explicit ParameterVariant(const std::string & name, const std::string & string_value)
   : name_(name)
   {
     value_.string_value = string_value;
     value_.parameter_type = rcl_interfaces::ParameterType::PARAMETER_STRING;
   }
-  ParameterVariant(const std::string & name, const std::vector<int8_t> & bytes_value)
+  explicit ParameterVariant(const std::string & name, const std::vector<int8_t> & bytes_value)
```

Comment:
It should indeed be `uint8` to follow our message parser which defines all the types: https://github.com/ros2/rosidl/blob/master/rosidl_parser/rosidl_parser/__init__.py#L484-L485

Probably the `ParameterValue` message needs to be updated to change `bytes_value` to `int8[]`.